tests/test_commands.py

```diff
@@ -101,6 +101,49 @@ def test_sort_dependencies_after_calling_another_command(
         assert sorted_pyproject_content == expected_pyproject_content
 
 
+@pytest.mark.parametrize(
+    ("argv", "command_path", "input_fixture", "expected_output", "expected_rc"),
+    (
+        (
+            ["", "check"],
+            "poetry.console.commands.check.CheckCommand",
+            "pyproject_multiple_groups.toml",
+            "pyproject_multiple_groups.toml",
+            1,
+        ),
+        (
+            ["", "check"],
+            "poetry.console.commands.check.CheckCommand",
+            "pyproject_multiple_groups__sorted.toml",
+            "pyproject_multiple_groups__sorted.toml",
+            0,
+        ),
+    ),
+)
+def test_sort_on_check_command(
+    application_factory,
+    fixture_dir,
+    poetry_from_fixture,
+    argv: list[str],
+    command_path: str,
+    input_fixture: str,
+    expected_output: str,
+    expected_rc: int,
+):
+    """
+    Makes sure that dependencies sort check runs on `poetry check`
+    """
+    poetry = poetry_from_fixture(input_fixture)
+    app = application_factory(poetry)
+
+    rc = app.run(input=ArgvInput(argv))
+    sorted_pyproject_content = poetry.file.path.read_text()
+    expected_pyproject_content = (fixture_dir / expected_output).read_text()
+
+    assert sorted_pyproject_content == expected_pyproject_content
+    assert rc == expected_rc
```

Comment:
I don't know why but i still get `rc = 0` instead of the expected `1`. I've not tested outside of pytest yet, i will try.

Reply:
If i run my fork in pre-commit i correctly get a `Dependencies are not sorted in tool.poetry.dependencies.` message, but it does not seem to set the `self._success` value to `False`... 🤔

Reply:
Ok, it looks like it is because `SortDependenciesPlugin.sort_dependencies(...)` isn't supposed to return a status code. I can workaround by raising a SystemExit, it works, but not sure that's the intended way for a plugin to work.

Reply:
You're right. `SortDependenciesPlugin.sort_dependencies(...)` is a just handler of TERMINATE event, which occurs after executing an original command (`CheckCommand`, `AddCommand`, etc). The exit code you're observing is returned by `CheckCommand.handler(..)`, and I'm not sure plugins are allowed to change the exit codes.

Raising `SystemExit` is quite a rough way to exit the interpreter. Doing so interrupts the "normal execution flow" of poetry and prevents necessary cleanup, the execution of other plugins\event handlers, and so on. I don't want the plugin to disrupt poetry

Reply:
Seems that `event` object (an instance of `cleo.events.console_terminate_event.ConsoleTerminateEvent`) keeps the exit code. Try to look at `event.set_exit_code(1)` as an alternative of exit(1)

Reply:
`event.set_exit_code(1)` works!

But there are other problems:
1. If pyproject.toml is valid and dependencies are not sorted, CheckCommand prints `All set!`. This is a little confusing.
2. If pyproject.toml is invalid and dependencies are not sorted, poetry doesn't trigger TERMINATE event, and so the plugin isn't called

Due to the latter, checking dependencies in `poetry check` is pointless

![image](https://github.com/andrei-shabanski/poetry-plugin-sort/assets/2552828/32d1a65d-66ba-48ea-83b9-fd052a3b6fcd)

Reply:
Ok then, let's close this 🤷